modules/widgets/social-icons.php

```diff
@@ -0,0 +1,500 @@
+		</p>
+
+		<?php
+			$support     = 'https://en.support.wordpress.com/widgets/social-media-icons-widget/#available-icons';
+			if ( 'es' === get_locale() ) {
+				$support = 'https://es.support.wordpress.com/social-media-icons-widget/#iconos-disponibles';
+			}
+			if ( 'pt-br' === get_locale() ) {
+				$support = 'https://br.support.wordpress.com/widgets/widget-de-icones-sociais/#ícones-disponíveis';
+			}
+		?>
+
+		<p>
+			<em><a href="<?php echo esc_url( $support ); ?>" target="_blank">
+				<?php esc_html_e( 'View available icons', 'jetpack' ); ?>
+			</a></em>
+		</p>
+
+	<?php
+	}
+
+	/**
+	 * Generates template to add icons.
+	 *
+	 * @param array $args Template arguments
+	 */
+	static function render_icons_template( $args = array() ) {
+		$defaults = array(
+			'url-icon-id'   => '',
+			'url-icon-name' => '',
+			'url-value'     => '',
+		);
+
+		$args = wp_parse_args( $args, $defaults );
+		?>
+
+		<div class="jetpack-social-icons-widget-item">
+			<div class="jetpack-social-icons-widget-item-wrapper">
+				<div class="handle"></div>
+
+				<p class="jetpack-widget-social-icons-url">
+					<?php
+						printf( '<input class="widefat id="%1$s" name="%2$s[]" type="text" placeholder="%3$s" value="%4$s"/>',
+							esc_attr( $args['url-icon-id'] ),
+							esc_attr( $args['url-icon-name'] ),
+							esc_attr__( 'Account URL', 'jetpack' ),
+							esc_url( $args['url-value'] )
+						);
+					?>
+				</p>
+
+				<p class="jetpack-widget-social-icons-remove-item">
+					<a class="jetpack-widget-social-icons-remove-item-button" href="javascript:;">
+						<?php esc_html_e( 'Remove', 'jetpack' ); ?>
+					</a>
+				</p>
+			</div>
+		</div>
+
+		<?php
+	}
+} // Jetpack_Widget_Social_Icons
+
+/**
+ * Register and load the widget.
+ *
+ * @access public
+ * @return void
+ */
+function jetpack_widget_social_icons_load() {
+	register_widget( 'Jetpack_Widget_Social_Icons' );
+}
+add_action( 'widgets_init', 'jetpack_widget_social_icons_load' );
+
+if ( ! function_exists( 'jetpack_social_menu_get_svg' ) ) :
+/**
+ * Return SVG markup.
+ *
+ * @param array $args {
+ *     Parameters needed to display an SVG.
+ *
+ *     @type string $icon  Required SVG icon filename.
+ * }
+ * @return string SVG markup.
+ */
+function jetpack_social_menu_get_svg( $args = array() ) {
+	// Make sure $args are an array.
+	if ( empty( $args ) ) {
+		return esc_html__( 'Please define default parameters in the form of an array.', 'jetpack' );
+	}
+
+	// Set defaults.
+	$defaults = array(
+		'icon'     => '',
+		'fallback' => false,
+	);
+
+	// Parse args.
+	$args = wp_parse_args( $args, $defaults );
+
+	// Define an icon.
+	if ( false === array_key_exists( 'icon', $args ) ) {
+		return esc_html__( 'Please define an SVG icon filename.', 'jetpack' );
+	}
+
+	// Set aria hidden.
+	$aria_hidden = ' aria-hidden="true"';
+
+	// Begin SVG markup.
+	$svg = '<svg class="icon icon-' . esc_attr( $args['icon'] ) . '"' . $aria_hidden . ' role="img">';
+
+	/*
+	 * Display the icon.
```

Comment:
Note: So this just prints a reference to the icon that is elsewhere in the DOM.  Doesn't read the file in or anything.